src/Castle.Core/Compatibility/RuntimeReflectionExtensions.cs

```diff
@@ -27,6 +27,16 @@ public static InterfaceMapping GetRuntimeInterfaceMap(this Type type, Type inter
 		{
 			return type.GetInterfaceMap(interfaceType);
 		}
+
+		public static Type CreateTypeInfo(this Emit.TypeBuilder typeBuilder)
+		{
+			return typeBuilder.CreateType();
+		}
+
+		public static Type AsType(this Type type)
```

Comment:
I had the same extension method at one point and then removed it because I didn't really like having the `AsType` extension method show up everywhere on `Type`, especially that the code was only used once in the whole library code (and now once in the unit tests). Thoughts?

https://github.com/castleproject/Core/blob/a2c67a06f1f7d0a7ef0032a1dbc9bc15de1a9792/src/Castle.Core/DynamicProxy/Generators/Emitters/AbstractTypeEmitter.cs#L330-L334